Bottom: 30fc01a2d89f2e52c05610db8f94e1b1a67473bc
Top:    f20bec5344a05ac5eb874799207dffb644550776
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 17:58:56 +0100

Refresh of change-penalty-function-in-rtr

---

diff --git a/src/rtree.c b/src/rtree.c
index c444e61..59793f8 100644
--- a/src/rtree.c
+++ b/src/rtree.c
@@ -878,21 +878,20 @@ contained (struct rtree_node *node, const BoxType * query)
 }
 
 
-static inline bigun
+static inline double
 penalty (struct rtree_node *node, const BoxType * query)
 {
-  long long score;
+  double score;
 
   /* Compute the area penalty for inserting here and return.
    * The penalty is the increase in area necessary
    * to include the query->
    */
-  score = (MAX (node->box.X2, query->X2) - MIN (node->box.X1, query->X1));
-  score *=
-    (MAX (node->box.Y2, query->Y2) - MIN (node->box.Y1, query->Y1));
+  score  = (MAX (node->box.X2, query->X2) - MIN (node->box.X1, query->X1));
+  score *= (MAX (node->box.Y2, query->Y2) - MIN (node->box.Y1, query->Y1));
   score -=
-    ((long long) node->box.X2 -
-     node->box.X1) * ((long long) node->box.Y2 - node->box.Y1);
+    (double)(node->box.X2 - node->box.X1) *
+    (double)(node->box.Y2 - node->box.Y1);
   return score;
 }
 
@@ -957,7 +956,7 @@ __r_insert_node (struct rtree_node *node, const BoxType * query,
     {
       int i;
       struct rtree_node *best_node;
-      long long score, best_score;
+      double score, best_score;
 
       if (force)
         {
